src/database/schema/ChatSchema.ts

```diff
@@ -354,29 +366,67 @@ class ChatSchema implements BaseDbSchema {
     }
   };
 
-  static async deleteChat(localDb: SQLiteDatabase, messageId: string): Promise<ChatSchema> {
-    const existingChat = await ChatSchema.getByid(localDb, messageId);
-    return new ChatSchema({
-      ...existingChat,
-      type: 'deleted',
-      message: 'This message has been deleted'
-    });
-  }
-
   static updateDeletedChatType = async (db: SQLiteDatabase, messageId: string) => {
     try {
+      const selectQuery = `SELECT raw_json FROM ${ChatSchema.getTableName()} WHERE id = ?;`;
+      const resultSet = await db.executeSql(selectQuery, [messageId]);
+      const result = resultSet[0].rows.item(0);
+      const rawJson = JSON.parse(result.raw_json);
+      rawJson.message_type = MESSAGE_TYPE_DELETED;
+      rawJson.text = DELETED_MESSAGE_TEXT;
+      const updatedRawJson = JSON.stringify(rawJson);
+
       const updateQuery = `UPDATE ${ChatSchema.getTableName()}
-        SET type = ?, message = ?
+        SET type = ?, message = ?, raw_json = ?
         WHERE id = ?;`;
 
-      const updateReplacement = ['deleted', 'This message has been deleted', messageId];
+      const updateReplacement = ['deleted', DELETED_MESSAGE_TEXT, updatedRawJson, messageId];
 
       await db.executeSql(updateQuery, updateReplacement);
     } catch (e) {
       console.log('error updating deleted chat:', e);
     }
   };
 
+  static updateDeletedRepliedChat = async (
+    db: SQLiteDatabase,
+    channelId: string,
+    messageId: string,
+    createdAt: string
+  ) => {
+    try {
+      const selectQuery = `SELECT id, raw_json FROM ${ChatSchema.getTableName()} WHERE channel_id = ? AND created_at > ?;`;
+      const resultSet = await db.executeSql(selectQuery, [channelId, createdAt]);
+      const {rows} = resultSet[0];
+
+      const updatePromises: Promise<any>[] = [];
+
+      // eslint-disable-next-line no-plusplus
+      for (let i = 0; i < rows.length; i++) {
+        const result = rows.item(i);
+        const rawJson = JSON.parse(result.raw_json);
+
+        // Check if messageId matches
+        if (rawJson.reply_data && rawJson.reply_data.id === messageId) {
+          const newJson = {...rawJson};
+          newJson.reply_data.text = DELETED_MESSAGE_TEXT;
+          newJson.message.reply_data.text = DELETED_MESSAGE_TEXT;
+          newJson.reply_data.message_type = MESSAGE_TYPE_DELETED;
+          newJson.updated = true;
+          const updatedRawJson = JSON.stringify(newJson);
+
+          // Update the chat
+          const updateQuery = `UPDATE ${ChatSchema.getTableName()} SET raw_json = ? WHERE id = ?;`;
+          updatePromises.push(db.executeSql(updateQuery, [updatedRawJson, result.id]));
+        }
+      }
+
+      await Promise.all(updatePromises);
+    } catch (e) {
+      console.log('error updating related chat:', e);
+    }
+  };
```

Comment:
<img src="https://avatars.githubusercontent.com/in/347564?s=41" alt="Image description" width="20" height="20">   CodeRabbit

The new method `updateDeletedRepliedChat` is introduced. It fetches all chat messages in a channel that were created after a certain time, checks if they are replies to a specific message, and if so, updates their `reply_data` fields. This change seems fine as long as the `raw_json` column is expected to contain a JSON string that can be parsed into an object with `reply_data` field.
```diff
+  static updateDeletedRepliedChat = async (
+    db: SQLiteDatabase,
+    channelId: string,
+    messageId: string,
+    createdAt: string
+  ) => {
+    try {
+      const selectQuery = `SELECT id, raw_json FROM ${ChatSchema.getTableName()} WHERE channel_id = ? AND created_at > ?;`;
+      const resultSet = await db.executeSql(selectQuery, [channelId, createdAt]);
+      const {rows} = resultSet[0];
+
+      const updatePromises: Promise<any>[] = [];
+
+      // eslint-disable-next-line no-plusplus
+      for (let i = 0; i < rows.length; i++) {
+        const result = rows.item(i);
+        const rawJson = JSON.parse(result.raw_json);
+
+        // Check if messageId matches
+        if (rawJson.reply_data && rawJson.reply_data.id === messageId) {
+          const newJson = {...rawJson};
+          newJson.reply_data.text = DELETED_MESSAGE_TEXT;
+          newJson.message.reply_data.text = DELETED_MESSAGE_TEXT;
+          newJson.reply_data.message_type = MESSAGE_TYPE_DELETED;
+          newJson.updated = true;
+          const updatedRawJson = JSON.stringify(newJson);
+
+          // Update the chat
+          const updateQuery = `UPDATE ${ChatSchema.getTableName()} SET raw_json = ? WHERE id = ?;`;
+          updatePromises.push(db.executeSql(updateQuery, [updatedRawJson, result.id]));
+        }
+      }
+
+      await Promise.all(updatePromises);
+    } catch (e) {
+      console.log('error updating related chat:', e);
+    }
+  };
```


<!-- This is an auto-generated comment by OSS CodeRabbit -->